src/serde/value/borrowed/de.rs

```diff
@@ -1048,4 +940,49 @@ mod test {
         let res: u128 = super::super::from_refvalue(&int128).expect("from_refvalue");
         assert_eq!(value, res);
     }
+
+    #[test]
+    fn variant() {
+        struct NameAndConfig<'v> {
+            name: String,
+            config: Option<borrowed::Value<'v>>,
+        }
+        impl<'v> serde::Deserialize<'v> for NameAndConfig<'v> {
+            fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+            where
+                D: serde::Deserializer<'v>,
+            {
+                #[derive(Deserialize)]
+                // This is ugly but it's needed for `serde::Deserialize` to not explode on lifetimes
+                #[serde(bound(deserialize = "'de: 'v, 'v: 'de"), untagged)]
```

Comment:
We'll forget the lifetime pinning hack in 2 weeks and will be puzzled in 6 months as to what this is doing and why
Also future rusts and/or serde's may be more ( or less ) strict here.
Perhaps add the compiler error when this bound annotation is omitted would suffice?